build-system/pr-check/checks.js

```diff
@@ -30,20 +30,35 @@ const {
   timedExecOrDie: timedExecOrDieBase,
 } = require('./utils');
 const {determineBuildTargets} = require('./build-targets');
+const {extensionBundles} = require('../compile/bundles.config');
 const {isPullRequestBuild} = require('../common/ci');
 const {reportAllExpectedTests} = require('../tasks/report-test-status');
 const {runNpmChecks} = require('./npm-checks');
 
 const FILENAME = 'checks.js';
 const timedExecOrDie = (cmd) => timedExecOrDieBase(cmd, FILENAME);
 
+function checkBundlesConfigExtensionBundles() {
+  let previousName = '';
+  for (const {name} of extensionBundles) {
+    if (name.localeCompare(previousName) < 0) {
+      throw new Error(
+        'extensionBundles in bundles.config.js should be alphabetically sorted by name'
+      );
+    }
+    previousName = name;
+  }
+}
+
```

Comment:
@rsimha

This check replaces the purpose of the previous call to `sort()` so that we can keep the diff stable.

This is probably not the proper way of adding it, so I'd appreciate assistance.

Reply:
If the list always has to be sorted, shouldn't the diff be relatively stable anyway?

Reply:
This is in case an extension is added manually (which happens all the time).

Reply:
Thanks for looping me in.

The intent is for `checks.js` and other files in `build-system/pr-check` to be wrapper scripts invoked during CI. They merely decide which `gulp` tasks to call and when to call them, and don't contain any actual test implementation. (I plan on refactoring / simplifying them further after the move to CircleCI.)

A couple options for this check:

1. **[Less trouble]** Move the contents of `checkBundlesConfigExtensionBundles()` back to `verifyExtensionBundles()`. It will get called whenever a build is performed, and serves as an implicit integrity check of `bundles.config.js`. (See line 143 below.)

https://github.com/ampproject/amphtml/blob/9b4be09e3ea24acf917602aa23100dab27f7e98a/build-system/tasks/extension-helpers.js#L132-L155

2. **[More trouble]** If you'd rather make this an explicit check that can be called independently during local development, move all of `verifyExtensionBundles()` to a new `gulp` task (e.g. `gulp verify-bundles`) and invoke the task in `checks.js`. We'll lose out on the implicit check during builds, but there will still be a safeguard during CI.

Here's a suggestion for option 1. (Feel free to optimize `colors.red()`, etc. to just `red()`.)
```diff
diff --git a/build-system/compile/bundles.config.js b/build-system/compile/bundles.config.js
index 831bdb8a3..13a71535f 100644
--- a/build-system/compile/bundles.config.js
+++ b/build-system/compile/bundles.config.js
@@ -1309,6 +1309,7 @@ function verifyBundle_(condition, field, message, name, found) {
 }
 
 exports.verifyExtensionBundles = function () {
+  const extensionNames = [];
   exports.extensionBundles.forEach((bundle) => {
     const bundleString = JSON.stringify(bundle, null, 2);
     verifyBundle_(
@@ -1359,5 +1360,20 @@ exports.verifyExtensionBundles = function () {
       bundle.name,
       bundleString
     );
+    extensionNames.push(bundle.name);
   });
+  const sorted = extensionNames.every((name, index) => {
+    return index === 0 || name >= extensionNames[index - 1];
+  });
+  if (!sorted) {
+    log(
+      colors.red('ERROR:'),
+      'Extensions in',
+      colors.cyan('extensionBundles'),
+      'in',
+      colors.cyan('build-system/compile/bundles.config.js'),
+      'are not alphabetically sorted.'
+    );
+    process.exit(1);
+  }
 };
```